### DISCUSSION TITLE
Generate new() or implement Default for generated structs

### DISCUSSION THREAD

**Discussion by wmmc88** [rust-lang/rust-bindgen#2469] [Q&A] [answered]
Is there an option to have Bindgen generate a `new` function or implement the `Default` for its generated structs? I have a lot of different large structs and explicitly initializing all the fields to 0 is very tedious.

**Comment by pvdrz** [comment, marked as answer]
You can use the [`derive_default`](https://docs.rs/bindgen/latest/bindgen/struct.Builder.html#method.derive_default) method or the `--with-derive-default` flag to derive `Default` for every type when possible.

Edit: Does this solve your issue @wmmc88?

**Reply by wmmc88** [reply]
Yup! Might be a good idea to make customizations like this more visible in the docs

**Reply by pvdrz** [reply]
Yeah, we have an issue with having way too many options right now :sweat_smile:. We're open to hear any suggestions about how to improve this.

**Reply by wmmc88** [reply]
I think it would be extremely helpful to put side by side comparisons in the bindgen book to visually show what each option does to change the bindings. I don't have a rust doc example, but I'd imagine something akin to what the [clangformat options page](https://clang.llvm.org/docs/ClangFormatStyleOptions.html) has
